### ISSUE TITLE
Widget list ever growing

### ISSUE THREAD

**Issue by jesserockz** [home-assistant/android#1108]
**Home Assistant Android version:** beta-477-495e022-full

**Android version:** 10

**Phone model:** Moto One Vision

**Home Assistant version:** 0.117.0b4

**Last working Home Assistant release (if known):**

**Description of problem:**

My launcher (nova) sometimes "looses" my layout/widgets. This means the internal list kept by the app is out of sync from the actual existing widgets with no way to delete.

**Traceback (if applicable):**

```

```

**Screenshot of problem:**
I only have 2 widgets on my home screen but list shows many.

![Screenshot_20201026-142708.png](https://user-images.githubusercontent.com/3060199/97125125-b939b700-1797-11eb-95ad-9929c1111b1b.png)

**Additional information:**
Maybe related: #1102 #1097 #1084

**Comment by dshokouhi** [comment]
So when the widgets get lost from the home screen the app does not get notified about it unlike when a user removes a widget we get notified.  The PR's referenced merely read the database and display the widgets we have saved.  I started looking into adding a delete button on the edit screen so users can clear them up as a user brought this up in Discord.  The app has no idea which ones are valid or not unfortunately.  As you create a widget we save it to the database and update as requested.
